### ISSUE TITLE
Enabled Addons are reset after minkube stop / start

### ISSUE THREAD

**Issue by alewis001** [kubernetes/minikube#6770]
* Minikube: `minikube v1.7.3 on Darwin 10.15.3`
* Hyperkit
* Kubernetes v1.17.3 on Docker 19.03.6

**The exact command to reproduce the issue**:

1. `minikube -p myprofile start`
   * The start output shows: `Enabling addons: default-storageclass, storage-provisioner`
2. `kubectl -n kube-system get pods`
   * metrics-server is not running, as expected.
3. `minikube -p myprofile addons enable metrics-server`
   * Output: `The 'metrics-server' addon is enabled`
4. `kubectl -n kube-system get pods`
   * Shows metrics-server running
5. `minikube -p myprofile addons list`
   * metrics-server appears enabled in the list, along with default-storageclass and storage-provisioner

At this point `$HOME/.minikube/profiles/myprofile/config.json` shows the metrics-server addon enabled in the Addons section.

1. `minikube -p myprofile stop`
2. `minikube -p myprofile start`
   * This command throws an error (see below).
   * At this point the Addons list in the `$HOME/.minikube/profiles/myprofile/config.json` is blanked out (`"Addons": null`).
   * The "Nodes.IP" value in the myprofile/config.json is also reset.
3. `minikube -p myprofile start`
   * A second attempt to start works.
   * The "Nodes.IP" value in the myprofile/config.json is set again, back to the value it had first time around.
4. `minikube -p myprofile addons list`
   * Metrics server will appear disabled
   * `kubectl -n kube-system get pods` shows metrics-server is running.
   * `kubectl top nodes` returns metrics.

**The full output of the command that failed**:<details>
The output from the start that fails:

😄  [bug2] minikube v1.7.3 on Darwin 10.15.3
✨  Using the hyperkit driver based on existing profile
⌛  Reconfiguring existing host ...
🔄  Starting existing hyperkit VM for "bug2" ...

💣  Unable to start VM. Please investigate and run 'minikube delete' if possible: post-start: command runner: ssh client: Error dialing tcp via ssh client: dial tcp 192.168.64.7:22: connect: no route to host

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose
</details>

**Comment by sharifelgamal** [comment]
I was able to reproduce this on 1.7.3. Thanks for reporting it! I'll take a closer look.

**Comment by alewis001** [comment]
Great. Glad it's not just specific to me but sorry for finding the issue :)

**Comment by tstromberg** [comment]
This appears to have been fixed during v1.8 at some time, I can't mimic it at v1.9.0-beta.1. Do you mind checking if it works for you now?

```
😄  [myprofile] minikube v1.9.0-beta.1 on Darwin 10.15.3
✨  Using the hyperkit driver based on existing profile
⌛  Reconfiguring existing host ...
🔄  Starting existing hyperkit VM for "myprofile" ...
🐳  Preparing Kubernetes v1.18.0-rc.1 on Docker 19.03.8 ...
🚀  Launching Kubernetes ... 
🌟  Enabling addons: default-storageclass, metrics-server, storage-provisioner
🏄  Done! kubectl is now configured to use "myprofile"
```

**Comment by alewis001** [comment]
Ok, thanks @tstromberg. I still see the same issue in 1.8.2 so I'll try updating to 1.9.0-beta-1 and report back.

**Comment by alewis001** [comment]
Great news! I can confirm that 1.9.0-beta.1 resolves both the issue in listing addons and the "start" failure (as mentioned in the details section of this issue).

Apart from using a new cluster, I also tried using a pre-existing one. Initially the addons list was wrong but I assume that's to be expected as minikube has lost track of what's enabled. I went through and re-enabled the addons. I assumed that from a cluster point of view this did nothing but from a minikube point-of-view, it corrected the list of enabled addons. Subsequent start/stop of the cluster reported the correct list of enabled plugins.

1.9.0 appears to be shaping up to be a corker! :) Thanks everyone involved for an awesome tool! Keep up the fantastic work.
